docs/root/configuration/http/http_filters/oauth2_filter.rst

```diff
@@ -106,13 +159,41 @@ And the below code block is an example of how we employ it as one of
     tls_context:
       sni: auth.example.com
 
+Finally, the following code block illustrates sample contents inside one of the two credential secrets.
+Both the :ref:`token_secret <envoy_v3_api_field_extensions.filters.http.oauth2.v3alpha.OAuth2Credentials.token_secret>`
+and the :ref:`hmac_secret <envoy_v3_api_field_extensions.filters.http.oauth2.v3alpha.OAuth2Credentials.hmac_secret>`
+can be defined in one shared file.
+
+.. code-block:: yaml
+
+  static_resources:
+    secrets:
+    - name: token
+      generic_secret:
+        secret: <Your secret value here>
+
+
 Notes
 -----
 
-This module does not currently provide much Cross-Site-Request-Forgery protection for the redirect
-loop to the OAuth server and back.
+When enabled, the OAuth filter does not protect against Cross-Site-Request-Forgery attacks on domains with
+cached authentication (in the form of cookies.)
+It is recommended to pair the OAuth filter with the :ref:`CSRF Filter <envoy_v3_api_msg_extensions.filters.http.csrf.v3.CsrfPolicy>`
+to prevent malicious social engineering.
+
+The service must be served over HTTPS for this filter to work properly, as the cookies use `;secure`. Without https, your
+:ref:`authorization_endpoint <envoy_v3_api_field_extensions.filters.http.oauth2.v3alpha.OAuth2Config.authorization_endpoint>`
+provider will likely reject the incoming request, and your access cookies will not be cached to bypass future logins.
+
+The signout path will redirect the current user to '/', and clear all authentication cookies related to
+the HMAC validation. Consequently, the OAuth filter will then restart the full OAuth flow at the root path,
+sending the user to the configured auth endpoint.
+
+:ref:`pass_through_matcher <envoy_v3_api_field_extensions.filters.http.oauth2.v3alpha.OAuth2Config.pass_through_matcher>` provides
+an interface for users to provide specific header matching criteria such that, when applicable, the OAuth flow is entirely skipped.
+When this occurs, the `oauth_success` metric is still incremented.
```

Comment:
Should we add an example for how to skip a path?

Reply:
The API documentation linked at the top should be enough.